### PR TITLE
chore: Add snippets for remote and local eval export

### LINKED ISSUE #3226: Provide an environment export to support Offline Mode
See [this related ticket](https://github.com/Flagsmith/flagsmith/issues/2024).

Now that we're supporting an increasingly larger number of SDKs operating in Offline Mode, it would be a real usability win if an `environment.json` file could be generated for our users which can then be manually imported into their SDKs for use in Offline Mode. 

@matthewelwell has some ideas on how to make this export really straight forward.

### PR DESCRIPTION
**Pull request by kyle-ssg**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Users can view how to export features for local and remote either by cURL or the CLI (Related CLI PR https://github.com/Flagsmith/flagsmith-cli/pull/19)

image

## How did you test this code?

- Tested new CLI
- Tested crud of api keys as this was edited

### CODE REVIEW

File: frontend/web/components/CodeHelp.js
Comment (zachaysan) [inline frontend/web/components/CodeHelp.js]: This area should be commented since the logic and underlying reasoning is non-obvious.

File: frontend/web/components/EnvironmentDocumentCodeHelp.tsx
Comment (zachaysan) [inline frontend/web/components/EnvironmentDocumentCodeHelp.tsx]: Is it possible to link to the appropriate Environment Settings page? Better than people clicking around looking for it.

**Review by zachaysan — COMMENTED** [review]
The code looks good to me, but JS isn't my forte. Hopefully the comments I raised will be useful.

File: frontend/web/components/EnvironmentDocumentCodeHelp.tsx
Comment (kyle-ssg) [reply frontend/web/components/EnvironmentDocumentCodeHelp.tsx]: My problem with doing that is it depends on the user's permissions, I'll do it but only if they are env admin

File: frontend/web/components/EnvironmentDocumentCodeHelp.tsx
Comment (matthewelwell) [inline frontend/web/components/EnvironmentDocumentCodeHelp.tsx]: The reasoning here was to support _offline_ mode which is kind of a subset of local evaluation mode. I think here we want to specify that, right?

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: frontend/web/components/EnvironmentDocumentCodeHelp.tsx
Comment (kyle-ssg) [reply frontend/web/components/EnvironmentDocumentCodeHelp.tsx]: Hmm we're mentioning this twice in the UI


image

Comment (matthewelwell) [reply frontend/web/components/EnvironmentDocumentCodeHelp.tsx]: Yeah, true, I guess it's just the tabs that concern me. I guess they are quite aligned with Remote / Local evaluation but not entirely. For example, you can use an offline handler as a default fall back in remote evaluation mode.

**Review by matthewelwell — COMMENTED** [review]
As discussed with @kyle-ssg on Slack, this is blocked waiting on the PR [here](https://github.com/Flagsmith/flagsmith-cli/pull/19) to be updated to separate the calls in the CLI for the different data structures.

**Comment by kyle-ssg** [comment]
> As discussed with @kyle-ssg on Slack, this is blocked waiting on the PR [here](https://github.com/Flagsmith/flagsmith-cli/pull/19) to be updated to separate the calls in the CLI for the different data structures.

CLI PR has been updated as well as this one to match the new --entity argument

**Review by zachaysan — COMMENTED** [review]
As far as I can tell this PR looks good, but I'm not strong enough with the JS approach to approve this PR as I have too many unknowns about this one.

**Comment by matthewelwell** [comment]
@kyle-ssg I've added some extra comments on the CLI PR.

**Comment by kyle-ssg** [comment]
@matthewelwell have resolved all of the comments in CLI PR

File: frontend/common/code-help/offline_client/offline-client-curl.js
Comment (matthewelwell) [inline frontend/common/code-help/offline_client/offline-client-curl.js]: nit: I'd argue this should be:

```suggestion
     -H 'x-environment-key: ${envId}' > flagsmith.json
```

I get that it's probably a very slim use case, but it's odd that it appends rather than replaces the existing data. Also the lack of white space annoyed me 😄

File: frontend/common/code-help/offline_server/offline-server-curl.js
Comment (matthewelwell) [inline frontend/common/code-help/offline_server/offline-server-curl.js]: As above. 

```suggestion
     -H 'x-environment-key: ${serversideEnvironmentKey}' > flagsmith.json
```

**Review by matthewelwell — APPROVED** [review]
Approved with a couple of minor comments.

File: frontend/common/code-help/offline_client/offline-client-curl.js
Comment (matthewelwell) [reply frontend/common/code-help/offline_client/offline-client-curl.js]: I've done this.

File: frontend/common/code-help/offline_server/offline-server-curl.js
Comment (matthewelwell) [reply frontend/common/code-help/offline_server/offline-server-curl.js]: I've done this.

**Comment by matthewelwell** [comment]
@kyle-ssg - I think this is ready to go once we've merged and released the [CLI PR](https://github.com/Flagsmith/flagsmith-cli/pull/19). Looks like that's a manual release process so I'll leave it to you tomorrow.

**Pull request merged**
